rhizome/host/lib/vm_setup.rb

```diff
@@ -83,8 +84,36 @@ def setup_networking(skip_persisted, gua, ip4, local_ip4, nics, ndp_needed, mult
     forwarding
   end
 
+  def manage_ip4_block_status(ip4, action)
+    ip_net = NetAddr::IPv4Net.parse(ip4).network.to_s
+    File.open(Util::NFTABLES_LOCK_FILE_PATH, File::RDWR | File::CREAT) do |lock|
+      lock.flock(File::LOCK_EX)
+      File.open(Util::NFTABLES_PATH, File::RDWR) do |nftfile|
+        nftfile.flock(File::LOCK_EX)
+
+        nft = nftfile.read
+        lines = nft.split("\n").map(&:strip)
+
+        commands = case action
+        when :unblock
+          lines.reject { _1.include?(ip_net) }
+        when :block
+          lines.insert(lines.index("type filter hook prerouting priority 0; policy accept;") + 1, "ip daddr #{ip_net} drop")
+        else
+          raise ArgumentError, "Invalid action: #{action}. Use :block or :unblock."
+        end
+
+        safe_write_to_file(Util::NFTABLES_PATH, commands.join("\n"))
```

Comment:
You don't need to open the file if you are using safe_write_to_file utility.

You still need lock file, because it is missing in safe_write_to_file, but I think we should add lock file logic to there instead of here (not in this PR though).

Reply:
But I need to be able to read the file, after locking it.

Reply:
ok, so you also read it. I'm bit conflicted about this pattern. We open it, read it and without closing we replace the inode through safe_write_to_file. What do you think about reading the file without a block;

```
nft_content = File.read(Util::NFTABLES_PATH)
```

Reply:
Random thought: can use of priority simplify, e.g. to superimpose accept rather than to generate a rule set all at priority zero?

Reply:
Yes, so that we wouldn't need to keep the file open while replacing it.